packages/ripple-binary-codec/src/binary.ts

```diff
@@ -16,26 +16,44 @@ import * as bigInt from 'big-integer'
  * Construct a BinaryParser
  *
  * @param bytes hex-string to construct BinaryParser from
+ * @param customDefinitions Rippled definitions used to parse the values of transaction types and such.
+ *                          Can be customized for sidechains and amendments.
  * @returns A BinaryParser
  */
-const makeParser = (bytes: string): BinaryParser => new BinaryParser(bytes)
+const makeParser = (
+  bytes: string,
+  customDefinitions: DefinitionContents = DEFINITIONS,
```

Comment:
nit (to apply elsewhere as well):
```suggestion
  definitions: DefinitionContents = DEFINITIONS,
```
The definitions aren't always custom.